ironfish/src/rpc/clients/socketClient.ts

```diff
@@ -34,10 +35,21 @@ export abstract class RpcSocketClient extends RpcClient {
   abstract client: IpcClient | net.Socket | null
   abstract isConnected: boolean
   abstract connection: Partial<RpcClientConnectionInfo>
+  rpcAuthToken: string
+
+  constructor(logger: Logger, rpcAuthToken: string) {
+    super(logger)
+    this.rpcAuthToken = rpcAuthToken
+  }
 
   abstract connect(options?: Record<string, unknown>): Promise<void>
   abstract close(): void
-  protected abstract send(messageId: number, route: string, data: unknown): void
+  protected abstract send(
+    messageId: number,
+    route: string,
+    data: unknown,
+    authToken: string,
```

Comment:
Technically we don't need to pass this through since any inheritors can access it, but this is fine. Though the type here is wrong because it should not be required.